src/tasks/qa.py

```diff
@@ -0,0 +1,174 @@
+"""Task definitions for question answering tasks."""
+import codecs
+import collections
+import logging as log
+import json
+import math
+import os
+import re
+from typing import Iterable, Sequence, List, Dict, Any, Type
+
+import torch
+import allennlp.common.util as allennlp_util
+from allennlp.training.metrics import Average, F1Measure
+from allennlp.data.token_indexers import SingleIdTokenIndexer
+from allennlp.data.fields import LabelField, MetadataField, ListField, SequenceLabelField
+from allennlp.data import Instance, Token
+
+from ..utils.data_loaders import process_sentence
+from ..utils.utils import truncate
+
+from .tasks import Task, sentence_to_text_field
+from .tasks import sentence_to_text_field, atomic_tokenize
+from .tasks import UNK_TOK_ALLENNLP, UNK_TOK_ATOMIC
+from .registry import register_task
+
+@register_task('multirc', rel_path='MultiRC/')
+class MultiRCTask(Task):
+    '''Multi-sentence Reading Comprehension task
+    See paper at https://cogcomp.org/multirc/ '''
+
+    def __init__(self, path, max_seq_len, name, **kw):
+        ''' '''
+        super().__init__(name, **kw)
+        self.scorer1 = F1Measure(positive_label=1)
+        self.scorer2 = Average() # to delete
+        self.scorer3 = F1Measure(positive_label=1)
+        self._score_tracker = collections.defaultdict(list)
+        self.val_metric = "%s_avg" % self.name
+        self.val_metric_decreases = False
+        self.max_seq_len = max_seq_len
+        self.files_by_split = {"train": os.path.join(path, "train.jsonl"),
+                               "val": os.path.join(path, "val.jsonl"),
+                               "test": os.path.join(path, "test.jsonl")}
+
+    def tokenizer_is_supported(self, tokenizer_name):
+        ''' For now, only support BERT '''
+        return "bert" in tokenizer_name
```

Comment:
What about this?

Reply:
Oops, this is deprecated and I'll delete it.